Neos.Neos/Migrations/Mysql/Version20210125134503.php

```diff
@@ -0,0 +1,56 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Persistence\Doctrine\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+use Neos\ContentRepository\Utility;
+
+final class Version20210125134503 extends AbstractMigration
+{
+    public function getDescription() : string
+    {
+        return 'add dimensions hash to node event model';
+    }
+
+    /**
+     * @param Schema $schema
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function up(Schema $schema) : void
+    {
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('ALTER TABLE neos_neos_eventlog_domain_model_event ADD dimensionshash VARCHAR(32) DEFAULT NULL');
+    }
+
+    /**
+     * @param Schema $schema
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function down(Schema $schema) : void
+    {
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('ALTER TABLE neos_neos_eventlog_domain_model_event DROP dimensionshash');
+    }
+
+    /**
+     * @param Schema $schema
+     * @throws \Doctrine\DBAL\Driver\Exception
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function postUp(Schema $schema): void
+    {
+        $eventLogResult = $this->connection->executeQuery('SELECT uid, dimension FROM neos_neos_eventlog_domain_model_event');
+
+        while ($eventLogInfo = $eventLogResult->fetchAssociative()) {
```

Comment:
This might need a loooong time for larger node event tables (and they tend to get pretty large). It thought about how to delegate more of the work to the database. What about: 

--psudo code--
```
while $dimensionsArray = 'SELECT dimension FROM neos_neos_eventlog_domain_model_event where dimensionshash IS NULL LIMIT 1' {
    $dimensionsHash = Utility::sortDimensionValueArrayAndReturnDimensionsHash($dimensionsArray);
    $this->connection->executeStatement('UPDATE neos_neos_eventlog_domain_model_event SET dimensionshash = ? WHERE dimension = ?', [$dimensionsHash, $dimensionsArray);
}
```
Should work in O(n), with n = dimensionCombinations and should be much faster.

Reply:
Hi @daniellienert , I adjusted the postUp function according to your suggestion. Can you please check, whether my adjustments are ok? Thank you.